bluesky/callbacks/zmq.py

```diff
@@ -0,0 +1,238 @@
+import ast
+import asyncio
+import multiprocessing
+import os
+import socket
+import time
+import zmq
+import zmq.asyncio
+from ..run_engine import Dispatcher, DocumentNames
+from ..utils import expiring_function
+
+
+class Publisher:
+    """
+    A callback that publishes documents to a 0MQ proxy.
+
+    Parameters
+    ----------
+    RE : ``bluesky.RunEngine``
+        RunEngine to which the Publisher will subscribe
+    address : string or tuple
+        Address of a running 0MQ proxy, given either as a string like
+        ``'127.0.0.1:5567'`` or as a tuple like ``('127.0.0.1', 5567)``
+
+    Example
+    -------
+
+    Publish to an existing Proxy at some known address.
+
+    >>> RE = RunEngine({})
+    >>> publisher = Publisher(RE, ('localhost', 5567))
+    """
+    def __init__(self, RE, address):
```

Comment:
Using a pattern like https://github.com/NSLS-II/GeRM/blob/master/pygerm/zmq.py#L51 can decouple this from which event loop it runs on.